### PR TITLE
Show documentation link if lint error links to documentation

### PR DESCRIPTION
**Pull request by philippfromme**
image

Related to https://github.com/camunda/camunda-modeler/issues/3609.

### CODE REVIEW

**Comment by philippfromme** [comment]
UX isn't great using the standard IBM Carbon icon at this size in combination with the subtle color change on hover. @YanaSegal What do you think? Any ideas?

https://github.com/camunda/camunda-modeler/assets/7633572/31d25977-2cfe-4322-9c76-0c85737a070f

File: client/src/app/panel/tabs/linting/LintingTab.less [hunk 1]
```diff
@@ -45,10 +45,27 @@
...
 
   .linting-tab-item__content {
-    display: flex;
     width: calc(100% - 200px);
     padding: var(--linting-tab-item-padding);
 
+    .linting-tab-item__link {
+      margin-left: 0.4em;
+      margin-right: 0.4em;
+      cursor: default;
```
Comment (barmac) [inline client/src/app/panel/tabs/linting/LintingTab.less:54]: Any reason why we would hide accessible cursor for links?

```suggestion
```

**Comment by nikku** [comment]
> UX isn't great using the standard IBM Carbon icon at this size in combination with the subtle color change on hover.

I suggest using the default link icon (cf. https://github.com/camunda/camunda-modeler/pull/3713#pullrequestreview-1518402908). Makes for good enough UX.

**Comment by YanaSegal** [comment]
@nikku @philippfromme I think we discussed showing the link icon when a user hovers over the line. for the link behavior - it depends what component you use - [a link](https://react.carbondesignsystem.com/?path=/story/components-link--paired-with-icon) or as [an icon button](https://react.carbondesignsystem.com/?path=/story/components-iconbutton--playground&args=kind:ghost)?

**Comment by nikku** [comment]
> I think we discussed showing the link icon when a user hovers over the line.

That sounds reasonable to me. It should be a link to not draw too much focus. Cf. also same pattern used in create/append menu: 

capture CJdzjG_optimized

File: client/src/app/panel/tabs/linting/LintingTab.less (hunk 1 above)
Comment (philippfromme) [reply client/src/app/panel/tabs/linting/LintingTab.less:54]: We're not consistent with cursors. We keep the default cursor for most clickable elements. 🤔

Comment (nikku) [reply client/src/app/panel/tabs/linting/LintingTab.less]: Let's distinguish links from buttons. Links more or less consistently show the default ("hand") cursor: 

capture Y1yLtC_optimized

**Comment by nikku** [comment]
It is not. It is core modeling, bpmn.io. A better version of the Carbon icon :wink:

**Review by barmac — APPROVED** [review]

**Review by YanaSegal — APPROVED** [review]

File: client/src/app/panel/tabs/linting/LintingTab.js [hunk 2]
```diff
@@ -124,6 +130,13 @@ function LintingTabItem(props) {
     </div>
     <div className="linting-tab-item__content">
       { message }
+      {
+        !isNil(documentationUrl) && <>
+          <a className="linting-tab-item__link" href={ documentationUrl } target="_blank" rel="noopener noreferrer" onClick={ stopPropagation }>
```
Comment (nikku) [inline client/src/app/panel/tabs/linting/LintingTab.js:135]: Can we add a descriptive title here such as "Open rule documentation"?

Comment (philippfromme) [reply client/src/app/panel/tabs/linting/LintingTab.js:135]: 👍🏻

Comment (philippfromme) [reply client/src/app/panel/tabs/linting/LintingTab.js]: https://github.com/camunda/camunda-modeler/pull/3713/commits/9d5951d68f2e8049debe831504d0c5a5684330ec 👀

**Comment by philippfromme** [comment]
The general concept is outlined here: https://miro.com/app/board/uXjVM5TE4Fg=/

**Review by nikku — APPROVED** [review]

**Comment by nikku** [comment]
I was just wondering if a simple `documentationRef` (we got it in element templates) would be sufficient. But I can get behind the idea of attaching potentially more things under `documentation`.

**Pull request merged**